app/serializers/user.js

```diff
@@ -21,4 +22,32 @@ export default class UserSerializer extends ApplicationSerializer {
     const links = { ...payload.links, first: null, last: null };
     return { data, links };
   }
+
+  normalizeResponse(store, primaryModelClass, payload, id, requestType) {
+    if (requestType === 'queryRecord' && payload.username) {
+      return {
+        data: {
+          id: payload.username,
+          type: primaryModelClass.modelName,
+          attributes: {
+            username: payload.username,
+          },
+        },
+      };
+    } else {
+      return super.normalizeResponse(
+        store,
+        primaryModelClass,
+        payload,
+        id,
+        requestType,
+      );
+    }
+  }
+
+  serialize() {
+    let json = super.serialize(...arguments);
+    delete json.id;
+    return json;
+  }
```

Comment:
What is this doing?

Reply:
As in payload id is also passed by ember-data  so I have delete id in payload

Reply:
Could you please add a reference? And what you're trying to do here

Reply:
in the code?